mdd/python/translation/openconfig_xr/xr_acls.py

```diff
@@ -0,0 +1,198 @@
+                if acl.type == 'oc-acl:ACL_IPV4':
+                    if not interface_cdb.ipv4.access_group.exists('ingress'):
+                        interface_cdb.ipv4.access_group.create('ingress')
+                    interface_cdb.ipv4.access_group['ingress'].name = acl.set_name
+                    self.log.info(f'{self.device_name} ACL {acl.set_name} added to interface {service_acl_interface.id} ingress')
+                elif acl.type == 'oc-acl-ext:ACL_IPV4_STANDARD':
+                    raise ValueError('XR does not support ACL type oc-acl-ext:ACL_IPV4_STANDARD.')
+
+
+def xr_acls_lines_program_service(self) -> None:
+    """
+    Program xr lines ingress and egress acls. Uses default template to configure lines.
+    """
+    device = self.root.devices.device[self.device_name].config
+    for service_line in self.service.oc_acl__acl.oc_acl_ext__lines.line:
+        if 'vty ' in service_line.id.lower():
+            matches = re.findall(r'[0-9]+', service_line.id)
+            if len(matches) == 2:
+                line_start = int(matches[0])
+                if line_start > 0:
+                    raise ValueError(f'XR VTY line configuration error. NSO-OC-Services XR uses the default line template which must start with line 0. You configured line {line_start}')
+                line_end = int(matches[1])
+                device.cisco_ios_xr__vty_pool.default.first_vty = 0
+                device.cisco_ios_xr__vty_pool.default.last_vty = line_end
+                if service_line.egress_acl_set:
+                    device.cisco_ios_xr__line.default.access_class.egress = service_line.egress_acl_set
+                if service_line.ingress_acl_sets.ingress_acl_set:
+                    for service_ingress in service_line.ingress_acl_sets.ingress_acl_set:
+                        device.cisco_ios_xr__line.default.access_class.ingress = service_ingress.ingress_acl_set_name
+                        if service_ingress.config.vrf and service_ingress.config.vrf != 'global':
+                            raise ValueError(
+                                f'XR VTY line configuration error. NSO-OC-Services XR only supports VTY line access-groups on the global routing instance at this time. You configured VRF {service_ingress.config.vrf}.')
+                        if service_ingress.config.vrf_also:
+                            raise ValueError(
+                                f'XR VTY line configuration error. NSO-OC-Services XR does not support vrf-also.')
+            else:
+                raise ValueError('line vty takes a start and an end line number range')
+
+
+def xr_acls_ntp_program_service(self) -> None:
+    """
+    Apply NTP ACLs
+    """
+    device = self.root.devices.device[self.device_name].config
+    if self.service.oc_acl__acl.oc_acl_ext__ntp.server.config.server_acl_set or self.service.oc_acl__acl.oc_acl_ext__ntp.peer.config.peer_acl_set:
+        device.cisco_ios_xr__ntp.access_group.delete()
+
+    # Serve
```

Comment:
I can't remember, but is serve a thing here or should it be server? I remember I did see "serve" as a legit value somewhere. If I'm wrong, ignore this.

Reply:
It is serve. I changed it in the model to server, but I was wrong.